rir/src/interpreter/builtins.cpp

```diff
@@ -3,6 +3,42 @@
 
 namespace rir {
 
+R_xlen_t asVecSize(SEXP x) {
+    if (isVectorAtomic(x) && LENGTH(x) >= 1) {
```

Comment:
I am confused by this, shouldn't it be `LENGTH(x) == 1` and otherwise give error?

Reply:
never mind, looked at the gnur implementation.. still weird though

Reply:
Ok I think it is broken after all. This passes as a fast path if length(x) > 1, but gnur gives an error in that case

> f <- pir.compile(rir.compile(function(x) vector("character", x)))
> 
> f(4)
[1] "" "" "" ""
> f(4:10)
[1] "" "" "" ""